### PR TITLE
Building Identification Agent

### PR DESCRIPTION
**Pull request by nagarajankarthik**
Pull request for building identification agent.

### CODE REVIEW

File: Agents/BuildingIdentificationAgent/README.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+The following optional parameters are only applicable if ```oneToMany``` is "true".
+- ```newTable``` (Optional): The name of the table in which the results of the matching should be stored. By default, this table is called 'matched_buildings' and it is created in the 'public' schema.
+- ```filterColumns``` (Optional): A JSONArray of strings specifying the columns of the user-specified table containing the attributes required to exclude unwanted geometries. This parameter
+- ```excludedValues``` (Optional): A JSONArray of JSONArrays. Each nested JSONArray contains a list of strings specifying values of the corresponding attribute specified in ```filterColumns``` which should be excluded from the matching.
+- ```mapping``` (Optional): The name of an Ontop mapping file specifying the triples linking the user-specified geometries to the buildings to which they are matched. This mapping file must be placed in the  'BuildingIdentificationAgent/src/resources' folder.
```
Comment (kok-foong) [inline Agents/BuildingIdentificationAgent/README.md:79]: I believe this needs to be updated

File: Agents/BuildingIdentificationAgent/BuildingIdentificationAgent/src/main/java/uk/ac/cam/cares/jps/agent/buildingidentification/BuildingIdentificationAgent.java [hunk 2]
```diff
@@ -0,0 +1,541 @@
...
+
+            // Properties of database containing buildings data.
+            String dbUrl = null;
+            String dbUser = null;
+            String dbPassword = null;
+
+            if (requestParams.has("dbUrl")) {
+                dbUrl = requestParams.getString("dbUrl");
+                dbUser = requestParams.getString("dbUser");
+                dbPassword = requestParams.getString("dbPassword");
```
Comment (kok-foong) [inline Agents/BuildingIdentificationAgent/BuildingIdentificationAgent/src/main/java/uk/ac/cam/cares/jps/agent/buildingidentification/BuildingIdentificationAgent.java:70]: I hope this is just for testing? Passing passwords in HTTP requests is not a good idea

File: Agents/BuildingIdentificationAgent/BuildingIdentificationAgent/src/main/java/uk/ac/cam/cares/jps/agent/buildingidentification/BuildingIdentificationAgent.java [hunk 3]
```diff
@@ -0,0 +1,541 @@
...
+                            "(SELECT ogc_fid, public.ST_TRANSFORM(\"%s\", %d) AS geometry FROM \"%s\") r1 " +
+                            "LEFT JOIN LATERAL ( " +
+                            " SELECT uuid, " +
+                            " r1.geometry <-> footprint_geometry AS dist " +
+                            " FROM building_footprints " +
+                            " ORDER BY dist " +
+                            " LIMIT 1 " +
+                            " ) r2 ON true " +
+                            " ) match " +
+                            " where \"%s\".ogc_fid = match.ogc_fid " +
```
Comment (kok-foong) [inline Agents/BuildingIdentificationAgent/BuildingIdentificationAgent/src/main/java/uk/ac/cam/cares/jps/agent/buildingidentification/BuildingIdentificationAgent.java:377]: No action needed, just looks susceptible to SQL injection

File: Agents/BuildingIdentificationAgent/BuildingIdentificationAgent/src/main/java/uk/ac/cam/cares/jps/agent/buildingidentification/BuildingIdentificationAgent.java (hunk 2 above)
Comment (nagarajankarthik) [reply Agents/BuildingIdentificationAgent/BuildingIdentificationAgent/src/main/java/uk/ac/cam/cares/jps/agent/buildingidentification/BuildingIdentificationAgent.java]: I have removed this part of the code, which was initially included for testing purposes.

**Review by kok-foong — APPROVED** [review]

**Pull request merged**